design.md

```diff
@@ -0,0 +1,38 @@
+## Montagu DB backup - design
+
+Our primary database sits on `production.montagu.dide.ic.ac.uk` on port 5432 (behind the VPN).  It uses streaming replication to stream to two [barman](https://www.pgbarman.org/) instances - one on `annex.montagu.dide.ic.ac.uk`, the other on AWS.
+
+Two concepts are important:
+
+1. a **base backup** is a snapshot of the postgres database at some point in time
+2. the **WAL (Write Ahead Log)** which can be replayed from a base backup to the present to restore the current database state
+
+The overall process is described in diagram form [here](https://github.com/vimc/montagu-machine/blob/master/docs/diagrams/Database%20backup.png)
+
+### The restore path
+
+The basic approach (without any real reference to the wrinkles that will appear below) is:
+
+1. On annex, `barman` dumps out a recovery directory (this is a base backup plus the WAL) using `barman recover` into the `montagu_db_volume` volume - this is our "nightly backup" and is controlled by cron so is error prone, and may stop running without warning.
+2. On annex, `bb8` ships this into the starport using `bb8 backup`, updating the directory on disk `~/starport/barman_to_starport` (this is the only backup target _from_ annex, even though everything else gets sent there).
```

Comment:
Add migration step - migrating to restore volume on annex from which Science is restored (and Production could be)..